ProjectBank/Core/IProjectRepository.cs

```diff
@@ -1,5 +1,12 @@
 namespace ProjectBank.Core;
-
-public class IProjectRepository
+public interface IProjectRepository
 {
+    Task<ProjectDTO> CreateProjectAsync(ProjectCreateDTO project);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadAllAsync();
+    Task<ProjectDTO> ReadProjectAsync(int projectId);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadProjectsByUserAsync(int userId);
+    Task<IReadOnlyCollection<ProjectDTO>> ReadProjectsByTag(int tagId);
```

Comment:
This should be renamed to "ReadProjectsByTagAsync" to reflect that it is a async method